### PR TITLE
T14452

### PR DESCRIPTION
**Pull request by katelk28**
Lots of style changes to Library and Thematic preset apps.

https://phabricator.endlessm.com/T14452

### CODE REVIEW

File: data/css/modules/documentCard/_base.scss [hunk 1]
```diff
@@ -1,16 +1,19 @@
 .DocumentCard {
     WebKitWebView {
         background-color: $background-light-color;
-        background-image: image-url('../templates/images/noise.png');
```
Comment (tchx84) [inline data/css/modules/documentCard/_base.scss]: Maybe this was needed in encyclopedia apps? Can't remember.

File: data/css/preset_a.scss [hunk 2]
```diff
@@ -22,77 +29,88 @@
...
-        font-size: 0.77em;
-        text-shadow: 0px 1px 0px white;
+    .ContentGroupContentGroup {
+        padding: 0px 100px 70px 100px;
     }
 }
 
-.set-page flowbox, .search-page flowbox {
-    margin-top: 20px;
+.set-page flowbox,
```
Comment (tchx84) [inline data/css/preset_a.scss]: Trailing white space here, please remove it (and all others). Usually, you can see these white spaces with ``` git log -p ```

File: data/widgets/card/knowledgeDocument.ui [hunk 3]
```diff
@@ -43,7 +43,7 @@
                 <property name="visible">True</property>
                 <property name="can_focus">False</property>
                 <property name="expand">True</property>
-                <property name="valign">center</property>
+                <property name="valign">top</property>
```
Comment (tchx84) [inline data/widgets/card/knowledgeDocument.ui]: 'top' does not exist, you probably meant something else. All of our apps show the following error:
```
Gtk-WARNING **: Failed to set property EknTableOfContents.valign to top: Could not parse enum: 'top'
```

File: js/app/widgets/tableOfContents.js [hunk 4]
```diff
@@ -415,7 +415,7 @@ const SectionButton = new Knowledge.Class({
         this.parent(params);
         let entry_class = Utils.get_element_style_class(TableOfContents, 'entry');
         this.get_style_context().add_class(entry_class);
-        section_title = section_title.toUpperCase();
+        // section_title = section_title.toUpperCase(); /* Don't want to delete yet, but this styling shouldn't be in the JS */
```
Comment (tchx84) [inline js/app/widgets/tableOfContents.js]: I noticed you removed this line on a later commit, why not doing it here? Please try a to unify when possible.

File: data/css/modules/card/_knowledgeDocument.scss [hunk 5]
```diff
@@ -30,58 +42,76 @@
...
         &:selected {
             color: white;
         }
 
-        &:selected {
-            background-color: transparentize(white, 1 - 0.247);
+        &:active {
+            color: transparentize(white, 1 - 0.90);
+            text-shadow: 1px 5px 3px rgba(0,0,0,0.2),
+                        0px -3px 30px rgba(255,255,255,0.1);
```
Comment (tchx84) [inline data/css/modules/card/_knowledgeDocument.scss]: Fix indentation please :)

File: data/css/modules/card/_knowledgeDocument.scss [hunk 6]
```diff
@@ -30,58 +42,76 @@
...
+        background-color: $toc-selected-color;
         border-radius: 50%;
-        border-width: 1px;
-        border-style: solid;
-        border-color: transparentize(black, 1 - 0.3);
+        margin: 1em 0em;
         padding: 5px;
-        box-shadow: inset 0px 1px 0px transparentize(white, 1 - 0.3);
+        box-shadow: 2px 8px 6px rgba(0,0,0,0.2),
+                 0px -5px 35px rgba(255,255,255,0.1);
```
Comment (tchx84) [inline data/css/modules/card/_knowledgeDocument.scss]: Same here and others places as well.

File: data/css/preset_b.scss [hunk 7]
```diff
@@ -4,25 +4,38 @@
...
 .home-page {
-    .LayoutBox {
+    .LayoutBox,
+    .ContentGroupContentGroup {
         min-height: 240px;
     }
 
     .ContentGroupContentGroup {
         background-color: black;
+        padding-top: 5px; /* FIXME:
```
Comment (tchx84) [inline data/css/preset_b.scss]: Same as with previous comments about comments :)

File: data/css/preset_b.scss [hunk 8]
```diff
@@ -65,25 +84,51 @@
...
-    .ContentGroupNoResultsMessage__subtitle {
-        font-size: 0.8em;
-        padding: 8px 15px;
-    }
 }
 
 .composite {
-    .home-page .LayoutBox {
-        min-height: 120px;
+    .home-page {
```
Comment (tchx84) [inline data/css/preset_b.scss]: Another trailing space.

File: data/preset/B.yaml [hunk 9]
```diff
@@ -29,8 +29,8 @@ defines:
         contents:
         - type: Banner.App
           properties:
-            valign: center
```
Comment (tchx84) [inline data/preset/B.yaml]: Please don't include changes that are not necessary :)

File: data/css/_mesh.scss [hunk 10]
```diff
@@ -11,11 +11,6 @@ $accent-dark-color: #d16550 !default;
 $legacy-post-card-border-color: transparentize(black, 1 - 0.70) !default;
 $legacy-post-card-border-color-hover: white !default;
 
-$body-font: 'Lato' !default;
-$title-font: 'Montserrat' !default;
-$display-font: 'Lato' !default;
-$display-font-composite: 'Roboto' !default;
```
Comment (tchx84) [inline data/css/_mesh.scss]: What happened to these values? The ones in typography are different, and I don't these values in preset_{a,b}.

File: data/css/_typography.scss [hunk 11]
```diff
@@ -1,5 +1,5 @@
-$title-font: 'Fira Sans' !default;
 $body-font: 'Merriweather' !default;
+$title-font: 'Fira Sans' !default;
```
Comment (tchx84) [inline data/css/_typography.scss]: Why is this necessary?

File: data/css/modules/card/_knowledgeDocument.scss [hunk 12]
```diff
@@ -49,8 +49,7 @@ $toc-align-padding: 15px;   /* This aligns the document title to the TOC
         background-color: transparent;
         border-radius: 1.75em;
         padding: 0.50em 0em;
-        text-shadow: 2px 8px 6px rgba(0,0,0,0.2),
```
Comment (tchx84) [inline data/css/modules/card/_knowledgeDocument.scss]: Please don't forget about these spaces between arguments.

File: data/css/modules/layout/_navigation.scss [hunk 13]
```diff
@@ -5,11 +5,29 @@
     &__backButton,
     &__forwardButton {
         color: white;
-        background-color: transparentize(black, 1 - 0.2);
-        border: 1px solid transparentize(white, 1 - 0.4);
-        box-shadow: inset 0px 1px 1px 0px transparentize(black, 1 - 0.5),
-            inset 0px 1px 4px 0px transparentize(black, 1 - 0.4);
+        background-color: transparentize(white, 1 - 0.2);
         -EknGrowButton-grow-pixels: 15;
+        // box-shadow: 2px 8px 6px rgba(0,0,0,0.2),
```
Comment (tchx84) [inline data/css/modules/layout/_navigation.scss]: hmm, maybe we should remove it and create a task for this issue instead?

File: data/widgets/card/title.ui [hunk 14]
```diff
@@ -33,7 +33,7 @@
             <property name="use_markup">True</property>
             <property name="wrap">False</property>
             <property name="wrap_mode">word-char</property>
-            <property name="max_width_chars">25</property>
+            <property name="max_width_chars">50</property>
```
Comment (tchx84) [inline data/widgets/card/title.ui]: Why is this changed needed? In any case, can you double check that this is not breaking other apps that use title card in composite mode?

File: js/app/widgets/tableOfContents.js [hunk 15]
```diff
@@ -415,7 +415,6 @@ const SectionButton = new Knowledge.Class({
         this.parent(params);
         let entry_class = Utils.get_element_style_class(TableOfContents, 'entry');
         this.get_style_context().add_class(entry_class);
-        // section_title = section_title.toUpperCase(); /* Don't want to delete yet, but this styling shouldn't be in the JS */
```
Comment (tchx84) [inline js/app/widgets/tableOfContents.js]: Please unify this change with the ones I previously mention.

File: data/css/modules/layout/_navigation.scss [hunk 16]
```diff
@@ -5,11 +5,29 @@
...
+        //          0px -5px 35px rgba(255,255,255,0.3);    
+                                                            /* FIXME: This should have
+                                                                a drop shadow, but it creates a GTK
+                                                                bug where the the button goes
+                                                                off the side of the app, specifically 
+                                                                when on an article page.
+                                                                Keeping commented.
+                                                            */
+
```
Comment (tchx84) [inline data/css/modules/layout/_navigation.scss]: HUGE trailing space here!

File: data/css/modules/card/_knowledgeDocument.scss [hunk 17]
```diff
@@ -154,10 +154,19 @@ $toc-align-padding: 15px;   /* This aligns the document title to the TOC
         &:hover {
             color: #54595a;
         }
-    }
+    }
```
Comment (tchx84) [inline data/css/modules/card/_knowledgeDocument.scss]: Please don't forget about trailing spaces :)

File: data/css/modules/card/_knowledgeDocument.scss [hunk 18]
```diff
@@ -154,10 +154,19 @@ $toc-align-padding: 15px;   /* This aligns the document title to the TOC
...
+    .InArticleSearch {
+        &__entry {
+            font-size: 21.1px;  /* 18 px spec * 88% because spec was at 110% scale not 125% */
+            background-color: transparentize(black, 1 - 0.8);
+        }
     }
+
+    /*
+    TableOfContents NOTE: 
+    Some TableOfContents composite styling is in the .TableOfContents class, as &--collapsed.
```
Comment (tchx84) [inline data/css/modules/card/_knowledgeDocument.scss]: It would be more helpful to comment this in the corresponding place where the styling is. I wonder why we have these separated.

File: data/css/preset_b.scss [hunk 19]
```diff
@@ -127,24 +129,34 @@
...
     .set-page,
-    .article-page {
-        .ContentGroupContentGroup {
-            background-color: transparentize(black, 1 - 0.8);
+    .search-page {
+        .content {
+            .Card__title,
+            .BannerSearch {
+                font-family: $title-font-composite;
+                font-size: 30px;
```
Comment (tchx84) [inline data/css/preset_b.scss]: Trailing space (looks like there are more).

**Comment by tchx84** [comment]
> library-list apps aren't broken for me! This is what I'm seeing:

@katelk28 did you rebuild the app after applying your changes? That is how I got to see that error on composite mode (if you don't rebuild, the app will still use the old css, before your changes).

File: data/css/modules/card/_knowledgeDocument.scss (hunk 18 above)
Comment (tchx84) [inline data/css/modules/card/_knowledgeDocument.scss]: Lets leave it there as a reminder we still need to revise how we are organizing selectors.

**Comment by katelk28** [comment]
@tchx84 
To the regression you noticed with this error... 
```
(com.endlessm.disabilities.en:27090): Gtk-WARNING **: Negative content width -49 (allocation 1, extents 50x0) while allocating gadget (node frame, owner GtkFrame)

(com.endlessm.disabilities.en:17150): Gtk-WARNING **: Negative content height -59 (allocation 1, extents 60x0) while allocating gadget (node frame, owner GtkFrame)
```
... I edited knowledgeDocument.js with some help from Watson; turns out that the toolbarframe was being allocated even when the table of contents isn't visible, so the code added just makes sure that it's not allocated. Then the sizing issue doesn't happen.


To this bug... 
```
*** BUG ***
In pixman_region32_init_rect: Invalid rectangle passed
Set a breakpoint on '_pixman_log_error' to debug
```
...I'm not sure what that is, and that is there on the master branch as well, so I don't think I caused that.
